src/Tokenizer/Token.php

```diff
@@ -156,7 +156,19 @@ public function clearChanged()
      */
     public function equals($other, $caseSensitive = true)
     {
-        $otherPrototype = $other instanceof self ? $other->getPrototype() : $other;
+        if ($other instanceof self) {
+            // inlined getPrototype() on this very hot path
+            if (!$other->isArray) {
```

Comment:
oh boy, i hate this "feature" of PHP tbh...

we basically access the internal, private property of some class, which works only because we run it from method of same class, yet different instance...
for me, it's violation of encapsulation...

maybe at least a comment documenting that we doing it and what for...

Reply:
I agree that this is not a feature which should be used all over the codebase.

but its a nice "shortcut" to squeeze out performance on very hot code-paths